### PR TITLE
Optimisation to fit atg model parameters.

### PR DESCRIPTION
**Pull request by miskosz**
Uses `scipy.optimize.least_squares` to find best fit.

I want to test on real data before merging.

[Technical] I am also considering optimising directly in the log space (as in finding log(A) and log(TG)). This could improve numerical stability of the optimisation since both A and TG are "multiplicative" parameters.

### CODE REVIEW

**Comment by miskosz** [comment]
Also I went fancy with the style, I hope that's ok :)

File: graphs/fit_atg_model.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
+from dataclasses import dataclass
+from typing import List
+
+import numpy as np
+import pytest
+from scipy.optimize import least_squares
+
+# TODO(miskosz): Pass in as a parameter?
+EXPONENT = 6.23
```
Comment (Aj0SK) [inline graphs/fit_atg_model.py:9]: I think we prefer optional parameter with default value and use argparse

File: graphs/fit_atg_model.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    """
+    a: float
+    tg: float
+
+    def predict(self, x: float) -> float:
+        ys = _model(params=[self.a, self.tg], xs=[x])
+        return ys[0]
+
+
+def fit_atg_model(xs: np.ndarray, ys: np.ndarray) -> AtgModelFit:
```
Comment (Aj0SK) [inline graphs/fit_atg_model.py:27]: I think I have to learn Python but this is good style :+1:

**Review by Aj0SK — APPROVED** [review]
I think it's okay after moving test elsewhere and... I like it :+1:

File: graphs/fit_atg_model.py (hunk 2 above)
Comment (lukipuki) [reply graphs/fit_atg_model.py:27]: Pozerám, že Mišo je v Pythone na míle pred nami. Ja používam podobný zápis v C++, napríklad `auto generate_age(std::mt19937* generator) -> uint32_t`.

File: graphs/fit_atg_model.py [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    with parameters `params` for the values `x` in `xs`, where `a=params[0]` and `tg=params[1]`.
+    """
+    a, tg = params
+    # Note(miskosz): Optimise `tg` in logspace if the following line becomes a problem.
+    assert tg > 0, f"No support for negative values for `tg` (tg={tg})."
+    return (a/tg) * (xs/tg)**EXPONENT * np.exp(-xs/tg)
+
+
+# Note(miskosz): Lousy test at the bottom of the file O:)
+# TODO(miskosz): Add a testing script. Atm can be tested by `pytest graphs/fit_atg_model.py`.
```
Comment (lukipuki) [inline graphs/fit_atg_model.py]: Chcelo by to normálne zabalíčkovať, máme na to issue: https://github.com/lukipuki/COVID-19-simulation/issues/4
Neviem, či sa k tomu dnes dostanem. Asi najprv spravím [CI](https://github.com/lukipuki/COVID-19-simulation/issues/7), potom by sme doňho pridali aj Python testy.

File: graphs/fit_atg_model.py [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+    NOTE: We compute the logarithm in logspace since the data spans multiple orders.
+    We don't care if the model is off by 10 for values close to 20000, but we care
+    for small values. We choose natural base of the logarithm.
+    """
+    return np.log(1. + _model(params=params, xs=xs)) - np.log(1. + ys)
+
+
+def _model(params: List[float], xs: np.ndarray) -> np.ndarray:
+    """
+    Returns predicted y-values of the model
```
Comment (lukipuki) [inline graphs/fit_atg_model.py:53]: Môžeš túto funkciu použiť tu? https://github.com/lukipuki/COVID-19-simulation/blob/3c70ab6011c5745944eea0ac3f0f4ab1e2ae7550/graphs/country.py#L37

File: graphs/fit_atg_model.py (hunk 1 above)
Comment (miskosz) [reply graphs/fit_atg_model.py:9]: This file is going to be imported, not invoked from a command line. It is a constant of the model though, the inelegant part is that now it lives in two files.

A little refactoring (coming soon) should solve the problem :)

File: graphs/fit_atg_model.py (hunk 3 above)
Comment (miskosz) [reply graphs/fit_atg_model.py]: Jj, ked bude CI tak tam mozem pridat python-specific veci.

File: graphs/fit_atg_model.py (hunk 4 above)
Comment (miskosz) [reply graphs/fit_atg_model.py:53]: Hej, zjednotim to nejak.. in an upcoming PR. Potrebujem totiz oddelit nacitavanie a prezentaciu dat v `country.py`.

**Comment by miskosz** [comment]
Ready for review. I scrapped the logspace optimisation, it was not giving results close to Bodova & Kollar (which is suspicious).

A tiez neviem ci po slovensky alebo anglicky, ale necham to anglicky keby nahodou Lukas recruitol niekoho zahranicneho :)

File: graphs/fit_atg_model.py (hunk 3 above)
Comment (lukipuki) [reply graphs/fit_atg_model.py]: FYI CI už je v pull requeste: https://github.com/lukipuki/COVID-19-simulation/pull/9

File: graphs/fit_atg_model.py (hunk 4 above)
Comment (lukipuki) [reply graphs/fit_atg_model.py:53]: :+1:

**Review by lukipuki — APPROVED** [review]

**Pull request merged**